src/py_sonar_scanner/__main__.py

```diff
@@ -17,19 +17,29 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
+import sys
 
 from py_sonar_scanner.configuration import Configuration
 from py_sonar_scanner.environment import Environment
+from py_sonar_scanner.logger import ApplicationLogger
 
 
 def scan():
-    cfg = Configuration()
-    cfg.setup()
+    log = ApplicationLogger.get_logger()
+    debug_enabled = False
+    try:
+        cfg = Configuration()
+        cfg.setup()
+        debug_enabled = cfg.is_debug()
 
-    env = Environment(cfg)
-    env.setup()
-    env.scanner().scan()
-    env.cleanup()
+        env = Environment(cfg)
+        env.setup()
+        env.scanner().scan()
+        env.cleanup()
```

Comment:
For execution consistency I have the feeling that `cleanup` in a `finally` block, as it essentially only deletes the scanner that was downloaded.